docs/custom-code/add-custom-dto.md

```diff
@@ -0,0 +1,243 @@
+---
+id: add-custom-dto
+title: How to add a custom DTO
+sidebar_label: Add a custom DTO
+slug: /custom-code/custom-dto
+---
+
+# How to add a custom DTO
+
+## General
+A DTO (Data Access Object) is an object that defines how the data will be sent over the network. 
+DTOs can be defined by TypeScript interfaces or by classes. Though, it is preferable to use classes, as TypeScript interfaces are removed during the the phase of translating TypeScript to JavaScript, and can't be referred an at runtime. However, classes are part of the JavaScript ES6 standard and are therefore preserved as real entities in the compiled JavaScript.
+
+The DTOs in Amplication are generated based on the relations between the entities and Prisma schema and types.
+Sometimes you have use-cases where you need DTOs supported by Prisma but not generated by Amplication yet.
+
+## Example
+The following example will demonstrate how to add a new custom DTO to your application and how to use it through the application instead of the base DTO.
+
+## Adding a new DTO
+1. Determine which DTO you need. For example, let's say you have an application with a Customer entity, and you want to filter all the customers with the `customerType` property - "Gold". On the generated app, you will get a DTO named `CustomerFindManyArgs` with a field (for GraphQL) or ApiProperty (for REST) from type: `CustomerWhereInput`.
```

Comment:
I think all this should go to a separate paragraph as part of the example and not as part of a step.
try to lay all the background for the example before you start with the instrcutions